src/lib.rs

```diff
@@ -98,6 +88,19 @@
 //! At this time, we do not recommend use for security-critical applications.
 //! Use at your own risk.
 //!
+//! # Useful features
+//!
+//! The main interfaces allow a [`Participant`] to process messages from
```

Comment:
> The main interfaces allow a [`Participant`] to process messages from other participants,

This is ambiguous to me. Instead of saying "the main interface" I think we should just say something like "A `Participant` processes messages from other `Participant`s. If the current sub-protocol is done, output values are produced."